fable-test-app/src/pages/Home.tsx

```diff
@@ -1,16 +1,74 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
 
 // Components (internal)
-import { DateModified, Heading, Text } from '../components';
+import { DateModified, Heading, NextHoliday, Text, Button } from '../components';
+import { holidayObject } from '../utils/constants';
 
 const Home: React.FC = () => {
+  const currentDate = new Date().getTime();
+  const [nextFederal, setNextFederal] = useState<holidayObject>();
+  const [nextNationwide, setNextNationwide] = useState<holidayObject>();
+
+  useEffect(() => {
+    axios.get('https://canada-holidays.ca/api/v1/holidays')
+    .then(({ data }) => {
+
+      // Assign next federal holiday
+      let fedAssigned = false;
+      data.holidays.map((holiday: holidayObject) => {
+        if (holiday.federal === 1) {
```

Comment:
Is it possible to combine the check for nationwide and federal holidays into this one iteration? Set the nationwide on the `else {` part? Are they usually the same too? If so, we can probably do just one object for it, no? And just have `<NextHoliday>` read the `holiday.federal` flag to render the holiday on whether it's a national holiday or a federal one, along with the `observedIn`

Reply:
The holidays won't normally be the same holiday, we are just entering the part of the year where the next 6 holidays will be the same. I tried to simplify it down a little bit.